### ISSUE TITLE
Bridging PRO, proteins, genes, transcripts

### ISSUE THREAD

**Issue by dosumis** [Cellular-Semantics/CL_KG#49]
STATUS: DRAFT

Use case:Queries that can unify Gene expression/function/markers whether they are proteins, genes or transcripts.

TBD:
 - Do we simply unify proteins/transcripts/genes or do we map all to a common gene node.  I think the latter.
 - How to deal with orthology mappings
 
The KG currently has has:

Proteins as PRO terms (Need mapping to Uniprot where possible - see https://github.com/obophenotype/cell-ontology/issues/2293).  These are used to record protein markers in CL.
Proteins as uniprot terms, currently used for GO annotation, originating in QuickGO.
Genes as ensembl IDs (Perhaps more strictly should be transcript IDs?)

We need unifying Gene nodes to support querying.  One potential way to achieve this is to defer to the translator standard as supported by Translator SRI:

https://github.com/TranslatorSRI/NodeNormalization/blob/master/documentation/NodeNormalization.ipynb

e.g. 
<img width="548" alt="image" src="https://github.com/user-attachments/assets/39f1f7c1-85d3-42a6-9440-158c4dabef7d">

Note - The uniprot ID here corresponds, I believe, to a generic uniprot ID used to refer to all protein products of a particular gene.  Keeping this as an 'Equivalent ID' on a unifying gene node might be problematic

Another possibility is to use mappings via MyGene.info.  In this case we need to make our own choices about what IDs are primary.

Approach:  

1. Draft schema along with test queries.
1. Break down and experiment with mapping tools

CC @dillerm - I think this is key to progress on a unified, queryable CL Knowledge Graph

We will break this down into separate steps:

- [x] #73
- [x] #76
    Open question: Should this translation happen at cellMark step or with other inputs or be part of ingestion.  Thinking about it - I think it makes more sense to be a general ingestion step  as we need to support ingestion of many IDs from many sources.

**Comment by dillerm** [comment]
> Note - The uniprot ID here corresponds, I believe, to a generic uniprot ID used to refer to all protein products of a particular gene. Keeping this as an 'Equivalent ID' on a unifying gene node might be problematic

I don't think it does correspond to all protein products of a single gene. Example: [CGRP](https://www.uniprot.org/uniprotkb/P06881/entry) and [calcitonin](https://www.uniprot.org/uniprotkb/P01258/entry). However, for genes that can encode for multiple gene products, wouldn't we want them to pull in all relevant protein products?

**Comment by dosumis** [comment]
Example outputs from Ensembl IDs coming from h5ad files --> CL_KG

```python
{ 'ensembl:ENSG00000169856': {'id': {'identifier': 'NCBIGene:3175',
   'label': 'ONECUT1'},
  'equivalent_identifiers': [{'identifier': 'NCBIGene:3175',
    'label': 'ONECUT1'},
   {'identifier': 'ENSEMBL:ENSG00000169856'},
   {'identifier': 'HGNC:8138', 'label': 'ONECUT1'},
   {'identifier': 'OMIM:604164'},
   {'identifier': 'UMLS:C1417951', 'label': 'ONECUT1 gene'},
   {'identifier': 'UniProtKB:Q9UBC0',
    'label': 'HNF6_HUMAN Hepatocyte nuclear factor 6 (sprot)'},
   {'identifier': 'PR:Q9UBC0', 'label': 'hepatocyte nuclear factor 6 (human)'},
   {'identifier': 'ENSEMBL:ENSP00000302630'},
   {'identifier': 'ENSEMBL:ENSP00000302630.4'}]
}
```

Uniprot IDs as input (Coming from GO annotation):

```python
{'UniProtKB:Q499E0': {'id': {'identifier': 'NCBIGene:215378',
   'label': 'Brinp3'},
  'equivalent_identifiers': [{'identifier': 'NCBIGene:215378',
    'label': 'Brinp3'},
   {'identifier': 'ENSEMBL:ENSMUSG00000035131'},
   {'identifier': 'MGI:2443035', 'label': 'Brinp3'},
   {'identifier': 'UniProtKB:Q499E0',
    'label': 'BRNP3_MOUSE BMP/retinoic acid-inducible neural-specific protein 3 (sprot)'},
   {'identifier': 'PR:Q499E0',
    'label': 'BMP/retinoic acid-inducible neural-specific protein 3 (mouse)'},
   {'identifier': 'ENSEMBL:ENSMUSP00000074201'},
   {'identifier': 'ENSEMBL:ENSMUSP00000074201.5'},
   {'identifier': 'ENSEMBL:ENSMUSP00000126074'},
   {'identifier': 'ENSEMBL:ENSMUSP00000126074.2'}],
  'type': ['biolink:Gene',
   'biolink:GeneOrGeneProduct',
   'biolink:GenomicEntity',
   'biolink:ChemicalEntityOrGeneOrGeneProduct',
   'biolink:PhysicalEssence',
   'biolink:OntologyClass',
   'biolink:BiologicalEntity',
   'biolink:ThingWithTaxon',
   'biolink:NamedThing',
   'biolink:PhysicalEssenceOrOccurrent',
   'biolink:MacromolecularMachineMixin',
   'biolink:Protein',
   'biolink:GeneProductMixin',
   'biolink:Polypeptide',
   'biolink:ChemicalEntityOrProteinOrPolypeptide'],
  'information_content': 95.4}}
```

**Comment by dosumis** [comment]
Pipeline - feed all IDs into normaliser. Use these to generate unified Gene IDs & make links to them.

Proposal:
* For proteins, we keep the protein node and bridge to translator standard en
* For Ensembl - unless the ID is for a specific transcript e.g. ensembl:ENSG00000169856.7 (in which case the node should be typed as transcript), we switch the primary gene node CURIE ID and label to NCBI:gene.  
* Gene nodes also store all the alternative IDs and labels (Schema on nodes TBD)

**Comment by dosumis** [comment]
Make intermediate dummy transcript node for Ensembl

**Comment by dosumis** [comment]
@dillerm  - Do you have a suggestion for this.  I've been trying but failing to find what to use in https://biolink.github.io/biolink-model/predicates.html

Maybe 

Gene  PRODUCES(RO:0003000)  Protein  (In NLM CL_KG Schema)
https://biolink.github.io/biolink-model/produces/

**Comment by dillerm** [comment]
Apologies, I overlooked this ping by accident. Yes, I think RO:0003000 would be my first suggestion to use here.

**Comment by dosumis** [comment]
Note  - we have NCBI_Gene IDs in the KG.  Full implementation of this ticket will unify these too.

**Comment by dosumis** [comment]
Note  - we have NCBI_Gene IDs in the KG.  Full implementation of this ticket will unify these too.
